src/django_components/component.py

```diff
@@ -672,14 +687,19 @@ def _render_impl(
             ):
                 self.on_render_before(context, template)
 
-                rendered_component = template.render(context)
-                new_output = self.on_render_after(context, template, rendered_component)
-                rendered_component = new_output if new_output is not None else rendered_component
+                # Get the component's HTML
+                html_content = template.render(context)
 
-        if is_dependency_middleware_active():
-            output = RENDERED_COMMENT_TEMPLATE.format(name=self.name) + rendered_component
-        else:
-            output = rendered_component
+                # Allow to optionally override/modify the rendered content
+                new_output = self.on_render_after(context, template, html_content)
+                html_content = new_output if new_output is not None else html_content
+
+                output = postprocess_component_html(
```

Comment:
This is where we hand off the rendered HTML to the logic that takes care of inserting component IDs and what not. Inside `postprocess_component_html` will be also where we will implement the scoped CSS feature, for example.